File: Demonstrator/Demonstrator.Models/DataModels/Patients/CarePlan.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;

namespace Demonstrator.Models.DataModels.Patients
{
    public class CarePlan
    {
        [BsonId]
        public ObjectId Id { get; set; }

        public bool InvolveFamilyOrCarer { get; set; }

        public string SignsFeelingUnwell { get; set; }

        public string PotentialTriggers { get; set; }

        public string WhatHelpsInCrisis { get; set; }

        public string EmergencyLocation { get; set; }

        public string EmergencyNumber { get; set; }

        public string CrisisNumber { get; set; }

        public bool PatientAcceptsPlan { get; set; }

        public DateTime PlanCreated { get; set; }

        public DateTime PlanUpdated { get; set; }

        public bool Active { get; set; }
    }
}

File: Demonstrator/Demonstrator.WebApp/AppSrc/core/models/CarePlan.ts
import { ValidationRules } from "aurelia-validation";
import { ICarePlan } from "../interfaces/ICarePlan";

export class CarePlan implements ICarePlan {
    constructor(public id?: string,
                public involveFamilyOrCarer?: boolean,
                public signsFeelingUnwell?: string,
                public potentialTriggers?: string,
                public whatHelpsInCrisis?: string,
                public emergencyLocation?: string,
                public emergencyNumber?: string,
                public crisisNumber?: string,
                public patientAcceptsPlan?: boolean,
                public planCreated?: Date,
                public planUpdated?: Date,
                public active?: boolean,
                public patientNhsNumber?: string) {
    }

    public carePlanRules() {
        const carePlanRules = ValidationRules
            .ensure('involveFamilyOrCarer').required()
            .ensure('signsFeelingUnwell').required()
            .ensure('potentialTriggers').required()
            .ensure('whatHelpsInCrisis').required()
            .ensure('patientAcceptsPlan').required()
            .rules;

        return carePlanRules;
    }
}